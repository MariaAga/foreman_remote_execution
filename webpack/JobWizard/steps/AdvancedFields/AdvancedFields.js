import React from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import { Form } from '@patternfly/react-core';
import { selectJobTemplate } from '../../JobWizardSelectors';
import {
  EffectiveUserField,
  TimeoutToKillField,
  PasswordField,
  KeyPassphraseField,
  EffectiveUserPasswordField,
  ConcurrencyLevelField,
  TimeSpanLevelField,
} from './Fields';

export const AdvancedFields = ({ advancedValue, setAdvancedValue }) => {
  const jobTemplate = useSelector(selectJobTemplate);
  const effectiveUser = jobTemplate.effective_user;
  return (
    <Form>
      {effectiveUser?.overridable && (
        <EffectiveUserField
          value={advancedValue.effectiveUserValue}
          setValue={newValue =>
            setAdvancedValue({
              effectiveUserValue: newValue,
            })
          }
        />
      )}
      <TimeoutToKillField
        value={advancedValue.timeoutToKill}
        setValue={newValue =>
          setAdvancedValue({
            timeoutToKill: newValue,
          })
        }
      />
      <PasswordField
        value={advancedValue.password}
        setValue={newValue =>
          setAdvancedValue({
            password: newValue,
          })
        }
      />
      <KeyPassphraseField
        value={advancedValue.keyPassphrase}
        setValue={newValue =>
          setAdvancedValue({
            keyPassphrase: newValue,
          })
        }
      />
      <EffectiveUserPasswordField
        value={advancedValue.effectiveUserPassword}
        setValue={newValue =>
          setAdvancedValue({
            effectiveUserPassword: newValue,
          })
        }
      />
      <ConcurrencyLevelField
        value={advancedValue.concurrencyLevel}
        setValue={newValue =>
          setAdvancedValue({
            concurrencyLevel: newValue,
          })
        }
      />
      <TimeSpanLevelField
        value={advancedValue.timeSpan}
        setValue={newValue =>
          setAdvancedValue({
            timeSpan: newValue,
          })
        }
      />
    </Form>
  );
};

AdvancedFields.propTypes = {
  advancedValue: PropTypes.object.isRequired,
  setAdvancedValue: PropTypes.func.isRequired,
};
export default AdvancedFields;
