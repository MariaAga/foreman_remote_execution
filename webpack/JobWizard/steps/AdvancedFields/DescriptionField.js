import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { FormGroup, TextInput, Button } from '@patternfly/react-core';
import { translate as __ } from 'foremanReact/common/I18n';

export const DescriptionField = ({ inputValues, value, setValue }) => {
  const generateDesc = () => {
    let newDesc = value;
    if (value) {
      const re = new RegExp('%\\{([^\\}]+)\\}', 'gm');
      const results = [...newDesc.matchAll(re)].map(result => ({
        name: result[1],
        text: result[0],
      }));
      results.forEach(result => {
        newDesc = newDesc.replace(result.text, inputValues[result.name]);
      });
    }
    return newDesc;
  };
  const [generatedDesc, setGeneratedDesc] = useState(generateDesc());
  const [isPreview, setIsPreview] = useState(true);

  const togglePreview = newValue => {
    setGeneratedDesc(generateDesc());
    setIsPreview(newValue);
  };

  const showPreview = (
    <Button variant="link" isInline onClick={() => togglePreview(true)}>
      {__('See the custom job description')}
    </Button>
  );
  const hidePreview = (
    <Button variant="link" isInline onClick={() => togglePreview(false)}>
      {__('Set the custom job description')}
    </Button>
  );
  return (
    <FormGroup
      label={__('Description')}
      fieldId="description"
      helperText={isPreview ? hidePreview : showPreview}
    >
      {isPreview ? (
        <TextInput id="description-preview" value={generatedDesc} isDisabled />
      ) : (
        <TextInput
          type="text"
          autoComplete="description"
          id="description"
          value={value}
          onChange={newValue => setValue(newValue)}
        />
      )}
    </FormGroup>
  );
};

DescriptionField.propTypes = {
  value: PropTypes.string,
  setValue: PropTypes.func.isRequired,
  inputValues: PropTypes.object.isRequired,
};
DescriptionField.defaultProps = {
  value: '',
};
