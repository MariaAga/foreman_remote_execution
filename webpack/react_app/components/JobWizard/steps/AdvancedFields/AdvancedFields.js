import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Form } from '@patternfly/react-core';
import {
  EffectiveUserField,
  TimeoutToKillField,
  PasswordField,
  KeyPassphraseField,
  EffectiveUserPasswordField,
  ConcurrencyLevelField,
  TimeSpanLevelField,
  TemplateInputsFields,
} from './Fields';

export const AdvancedFields = ({
  effective_user: effectiveUser,
  template_inputs_with_foreign: templateInputs,
  job_template: {
    execution_timeout_interval: executionTimeoutInterval,
    // description_format: descriptionFormat,
  },
  advancedValue,
  setAdvancedValue,
}) => {
  return (
    <Form>
      <TemplateInputsFields
        inputs={templateInputs}
        value={advancedValue.templateValues}
        setValue={newValue =>
          setAdvancedValue({ ...advancedValue, templateValues: newValue })
        }
      />
      {effectiveUser?.overridable && (
        <EffectiveUserField
          value={advancedValue.effectiveUserValue}
          setValue={newValue =>
            setAdvancedValue({ ...advancedValue, effectiveUserValue: newValue })
          }
        />
      )}
      <TimeoutToKillField
        value={advancedValue.timeoutToKill}
        setValue={newValue =>
          setAdvancedValue({ ...advancedValue, timeoutToKill: newValue })
        }
      />
      {/* <DescriptionField
        description={description}
        setDescription={setDescription}
        templateInputs={templateInputs}
        descriptionFormat={descriptionFormat}
      />
      <DescriptionTemplateField /> */}
      <PasswordField
        value={advancedValue.password}
        setValue={newValue =>
          setAdvancedValue({ ...advancedValue, password: newValue })
        }
      />
      <KeyPassphraseField
        value={advancedValue.keyPassphrase}
        setValue={newValue =>
          setAdvancedValue({ ...advancedValue, keyPassphrase: newValue })
        }
      />
      <EffectiveUserPasswordField
        value={advancedValue.effectiveUserPassword}
        setValue={newValue =>
          setAdvancedValue({
            ...advancedValue,
            effectiveUserPassword: newValue,
          })
        }
      />

      <ConcurrencyLevelField
        value={advancedValue.concurrencyLevel}
        setValue={newValue =>
          setAdvancedValue({ ...advancedValue, concurrencyLevel: newValue })
        }
      />
      <TimeSpanLevelField
        value={advancedValue.timeSpan}
        setValue={newValue =>
          setAdvancedValue({ ...advancedValue, timeSpan: newValue })
        }
      />
    </Form>
  );
};

AdvancedFields.propTypes = {
  effective_user: PropTypes.object,
  job_template: PropTypes.shape({
    execution_timeout_interval: PropTypes.string,
  }),
  template_inputs_with_foreign: PropTypes.array,
  advancedValue: PropTypes.object.isRequired,
  setAdvancedValue: PropTypes.func.isRequired,
};
AdvancedFields.defaultProps = {
  effective_user: {},
  job_template: {},
  template_inputs_with_foreign: [],
};
export default AdvancedFields;
