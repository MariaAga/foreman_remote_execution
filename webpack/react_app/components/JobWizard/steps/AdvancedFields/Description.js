import React, { useState } from 'react';
import PropTypes from 'prop-types';
import {
  Form,
  FormGroup,
  TextInput,
  TextArea,
  FormSelect,
  Checkbox,
  Popover,
  ActionGroup,
  Button,
  Radio,
} from '@patternfly/react-core';
import { HelpIcon, ExclamationCircleIcon } from '@patternfly/react-icons';

import { translate as __ } from 'foremanReact/common/I18n';

const generateDefaultDescription = (descriptionFormat, templateInputs) => {
  if (descriptionFormat) {
    return descriptionFormat;
  }
  const generatedDescription = '%{template_name}';
  if (templateInputs) {
    const inputs = templateInputs
      .map(input => input.name)
      .map(name => `${name}=\\"%{${name}}\\"`)
      .join(' ');
    return generatedDescription.concat(' ', `with inputs ${inputs}`);
  }
  return generatedDescription;
};

const DescriptionTemplateField = ({
  description,
  setDescription,
  disabled,
  descriptionFormat,
  templateInputs,
}) => {
  // const defaultValue = generateDefaultDescription(
  //   descriptionFormat,
  //   templateInputs
  // );
  // :value => f.object.description_format || job_template.generate_description_format,

  const helpBlock = (
    <p>
      {__('This template is used to generate the description.')}
      {__('Input values can be used using the syntax %{package}.')}{' '}
      {__(
        'You may also include the job category and templatename using %{job_category} and %{template_name}.'
      )}
    </p>
  );
  return (
    <FormGroup
      label={__('Description template')}
      labelIcon={helpLabel(helpBlock)}
      fieldId="description_format'"
      type="text"
    >
      <TextArea
        disabled={disabled}
        id="description_format"
        rows={2}
        value={description}
        setValue={setDescription}
      />
    </FormGroup>
  );
};
const DescriptionField = ({
  description,
  setDescription,
  disabled,
  descriptionFormat,
  templateInputs,
}) => {
  const [isChecked, setIsChecked] = useState(true);
  const defaultValue = generateDefaultDescription(
    descriptionFormat,
    templateInputs
  );
  const handleCheck = (checked, event) => {
    setIsChecked(checked);
    setDescription(defaultValue);
  };
  return (
    <FormGroup
      label={__('Description')}
      labelIcon={helpLabel(
        __(
          'A user to be used for executing the script. If it differs from the SSH user, su or sudo is used to switch the accounts.'
        )
      )}
      fieldId="description"
      type="text"
    >
      <TextArea
        readOnly={isChecked}
        disabled={disabled}
        id="description"
        rows={2}
        value={description}
        setValue={setDescription}
      />
      <Checkbox isChecked={isChecked} onChange={handleCheck} />
      {__('Use default description template')}
    </FormGroup>
  );
};
