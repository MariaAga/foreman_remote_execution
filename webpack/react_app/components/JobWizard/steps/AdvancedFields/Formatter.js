import React, { useEffect } from 'react';
import { useSelector } from 'react-redux';
import { FormGroup, TextInput } from '@patternfly/react-core';
import SearchBar from 'foremanReact/components/SearchBar';
import { helpLabel } from '../form/FormHelpers';
import { SelectField } from '../form/SelectField';

const TemplateSearchField = ({
  name,
  controller,
  labelText,
  required,
  defaultValue,
  setValue,
  values,
}) => {
  const searchQuery = useSelector(
    state => state.autocomplete[name]?.searchQuery
  );
  useEffect(() => {
    setValue({ ...values, [name]: searchQuery });
  }, [searchQuery]);
  return (
    <FormGroup
      key={name}
      label={name}
      labelIcon={helpLabel(labelText)}
      fieldId={name}
      isRequired={required}
    >
      <SearchBar
        initialQuery={defaultValue}
        data={{
          controller,
          autocomplete: {
            id: name,
            url: `/${controller}/auto_complete_search`,
            useKeyShortcuts: true,
          },
        }}
      />
    </FormGroup>
  );
};

export const formatter = (input, values, setValue) => {
  const isSelectType = !!input?.options;
  const inputType = input.value_type;
  const isTextType = inputType === 'plain' || !inputType; // null defaults to plain

  const { name, required, hidden_value: hidden } = input;
  const labelText = input.description;
  const defaultValue = input.default;
  const value = values[name] || defaultValue;

  if (isSelectType) {
    const options = input.options.split(/\r?\n/).map(option => option.trim());
    return (
      <SelectField
        key={name}
        isRequired={required}
        label={name}
        fieldId={name}
        options={options}
        labelText={labelText}
        value={value}
        setValue={newValue => setValue({ ...values, [name]: newValue })}
      />
    );
  }
  if (isTextType) {
    return (
      <FormGroup
        key={name}
        label={name}
        labelIcon={helpLabel(labelText)}
        fieldId={name}
        isRequired={required}
      >
        <TextInput
          className={hidden ? 'masked-input' : null}
          required={required}
          rows={2}
          id={name}
          type="text"
          value={value}
          onChange={newValue => setValue({ ...values, [name]: newValue })}
        />
      </FormGroup>
    );
  }
  if (inputType === 'date') {
    return <span key={name}>DateTime here</span>;
  }
  if (inputType === 'search') {
    const controller = input.resource_type;
    // TODO: get text from redux autocomplete
    return (
      <TemplateSearchField
        name={name}
        controller={controller}
        labelText={labelText}
        required={required}
        defaultValue={defaultValue}
        setValue={setValue}
        values={values}
      />
    );
  }

  return null;
};
