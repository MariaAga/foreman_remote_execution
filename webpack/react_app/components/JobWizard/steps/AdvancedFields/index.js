import React, { useEffect, useCallback } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { get } from 'foremanReact/redux/API';
import { AdvancedFields } from './AdvancedFields';
import { selectJobTemplate } from '../../JobWizardSelectors';
import { JOB_TEMPLATE } from '../../JobWizardConsts';

const ConnectedAdvancedFields = ({
  jobTemplateID,
  advancedValue,
  setAdvancedValue,
}) => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(
      get({
        key: JOB_TEMPLATE,
        url: `/ui_job_wizard/template/${jobTemplateID}`,
      })
    );
  }, [jobTemplateID, dispatch]);

  useEffect(() => setAdvancedValueDefaults, [
    jobTemplate,
    setAdvancedValueDefaults,
  ]);

  const jobTemplate = useSelector(selectJobTemplate);
  const setAdvancedValueDefaults = useCallback(
    () =>
      setAdvancedValue({
        ...advancedValue,
        effectiveUserValue: jobTemplate.effectiveUser?.value || '',
        timeoutToKill: jobTemplate.job_template?.executionTimeoutInterval || '',
      }),
    [setAdvancedValue, jobTemplate]
  );
  return (
    <AdvancedFields
      {...jobTemplate}
      advancedValue={advancedValue}
      setAdvancedValue={setAdvancedValue}
    />
  );
};

ConnectedAdvancedFields.propTypes = {
  jobTemplateID: PropTypes.number.isRequired,
};

export default ConnectedAdvancedFields;
