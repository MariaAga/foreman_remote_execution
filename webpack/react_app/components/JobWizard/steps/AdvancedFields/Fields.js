import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import { FormGroup, TextInput } from '@patternfly/react-core';
import { translate as __ } from 'foremanReact/common/I18n';
import { helpLabel } from '../form/FormHelpers';
import { formatter, isNumber } from './Formatter';

export const EffectiveUserField = ({ value, setValue }) => (
  <FormGroup
    label={__('Effective user')}
    labelIcon={helpLabel(
      __(
        'A user to be used for executing the script. If it differs from the SSH user, su or sudo is used to switch the accounts.'
      )
    )}
    fieldId="effective-user"
  >
    <TextInput
      autoComplete="effective-user"
      id="effective-user"
      type="text"
      value={value}
      onChange={newValue => setValue(newValue)}
    />
  </FormGroup>
);

export const TimeoutToKillField = ({ value, setValue }) => {
  const [validated, setValidated] = useState('default');
  const handleTextInputChange = newValue => {
    setValidated(isNumber(newValue) ? 'default' : 'error');
    setValue(newValue);
  };

  return (
    <FormGroup
      label={__('Timeout to kill')}
      labelIcon={helpLabel(
        __(
          'Time in seconds from the start on the remote host after which the job should be killed.'
        )
      )}
      fieldId="timeout-to-kill"
      validated={validated}
      helperTextInvalid={__('Timeout to kill has to be a number')}
    >
      <TextInput
        type="number"
        value={value}
        placeholder={__('For example: 1, 2, 3, 4, 5...')}
        autoComplete="timeout-to-kill"
        id="timeout-to-kill"
        validated={validated}
        onChange={handleTextInputChange}
      />
    </FormGroup>
  );
};

export const PasswordField = ({ value, setValue }) => (
  <FormGroup
    label={__('Password')}
    labelIcon={helpLabel(
      __(
        'Password is stored encrypted in DB until the job finishes. For future or recurring executions, it is removed after the last execution.'
      )
    )}
    fieldId="password"
  >
    <TextInput
      autoComplete="password"
      id="password"
      type="password"
      placeholder="*****"
      value={value}
      onChange={newValue => setValue(newValue)}
    />
  </FormGroup>
);

export const KeyPassphraseField = ({ value, setValue }) => (
  <FormGroup
    label={__('Private key passphrase')}
    labelIcon={helpLabel(
      __(
        'Key passphrase is only applicable for SSH provider. Other providers ignore this field. Passphrase is stored encrypted in DB until the job finishes. For future or recurring executions, it is removed after the last execution.'
      )
    )}
    fieldId="key-passphrase"
  >
    <TextInput
      autoComplete="key-passphrase"
      id="key-passphrase"
      type="password"
      placeholder="*****"
      value={value}
      onChange={newValue => setValue(newValue)}
    />
  </FormGroup>
);

export const EffectiveUserPasswordField = ({ value, setValue }) => (
  <FormGroup
    label={__('Effective user password')}
    labelIcon={helpLabel(
      __(
        'Effective user password is only applicable for SSH provider. Other providers ignore this field. Password is stored encrypted in DB until the job finishes. For future or recurring executions, it is removed after the last execution.'
      )
    )}
    fieldId="effective-user-password"
  >
    <TextInput
      autoComplete="effective-user-password"
      id="effective-user-password"
      type="password"
      placeholder="*****"
      value={value}
      onChange={newValue => setValue(newValue)}
    />
  </FormGroup>
);

export const ConcurrencyLevelField = ({ value, setValue }) => {
  const [validated, setValidated] = useState('default');
  const handleTextInputChange = newValue => {
    setValidated(isNumber(newValue) && newValue >= 1 ? 'default' : 'error');
    setValue(newValue);
  };
  return (
    <FormGroup
      label={__('Concurrency level')}
      labelIcon={helpLabel(
        __(
          'Run at most N tasks at a time. If this is set and proxy batch triggering is enabled, then tasks are triggered on the smart proxy in batches of size 1.'
        )
      )}
      fieldId="concurrency-level"
      validated={validated}
      helperTextInvalid={__(
        'Concurrency level has to be a number, and at least 1'
      )}
    >
      <TextInput
        type="number"
        autoComplete="concurrency-level"
        id="concurrency-level"
        placeholder={__('For example: 1, 2, 3, 4, 5...')}
        value={value}
        validated={validated}
        onChange={newValue => handleTextInputChange(newValue)}
      />
    </FormGroup>
  );
};

export const TimeSpanLevelField = ({ value, setValue }) => {
  const [validated, setValidated] = useState('default');
  const handleTextInputChange = newValue => {
    setValidated(isNumber(newValue) && newValue >= 1 ? 'default' : 'error');
    setValue(newValue);
  };
  return (
    <FormGroup
      label={__('Time span')}
      labelIcon={helpLabel(
        __(
          'Distribute execution over N seconds. If this is set and proxy batch triggering is enabled, then tasks are triggered on the smart proxy in batches of size 1.'
        )
      )}
      fieldId="time-span"
      validated={validated}
      helperTextInvalid={__(
        'Concurrency level has to be a number, and at least 1'
      )}
    >
      <TextInput
        min={1}
        type="number"
        autoComplete="time-span"
        id="time-span"
        placeholder={__('For example: 1, 2, 3, 4, 5...')}
        value={value}
        validated={validated}
        onChange={newValue => handleTextInputChange(newValue)}
      />
    </FormGroup>
  );
};

export const TemplateInputsFields = ({ inputs, value, setValue }) => {
  inputs = inputs?.filter(input => input.advanced);
  return <>{inputs?.map(input => formatter(input, value, setValue))}</>;
};

EffectiveUserField.propTypes = {
  value: PropTypes.string.isRequired,
  setValue: PropTypes.func.isRequired,
};
TimeoutToKillField.propTypes = EffectiveUserField.propTypes;
PasswordField.propTypes = EffectiveUserField.propTypes;
KeyPassphraseField.propTypes = EffectiveUserField.propTypes;
EffectiveUserPasswordField.propTypes = EffectiveUserField.propTypes;
ConcurrencyLevelField.propTypes = EffectiveUserField.propTypes;
TimeSpanLevelField.propTypes = EffectiveUserField.propTypes;

TemplateInputsFields.propTypes = {
  inputs: PropTypes.array.isRequired,
  value: PropTypes.object.isRequired,
  setValue: PropTypes.func.isRequired,
};
