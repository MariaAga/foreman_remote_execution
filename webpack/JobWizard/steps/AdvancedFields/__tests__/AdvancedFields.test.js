import React from 'react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import * as patternfly from '@patternfly/react-core';
import { mount } from '@theforeman/test';
import { AdvancedFields } from '../AdvancedFields';

jest.spyOn(patternfly, 'FormGroup');
patternfly.FormGroup.mockImplementation(props => (
  <div>{props.navAriaLabel}</div>
));
const mockStore = configureMockStore([]);
const store = mockStore({
  JOB_TEMPLATE: { response: { effective_user: { overridable: true } } },
});
describe('AdvancedFields', () => {
  it('rendring', () => {
    const component = mount(
      <Provider store={store}>
        <AdvancedFields advancedValues={{}} setAdvancedValues={jest.fn()} />
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
