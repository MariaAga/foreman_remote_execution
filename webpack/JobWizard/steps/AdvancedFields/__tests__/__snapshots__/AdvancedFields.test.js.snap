// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdvancedFields rendring 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <AdvancedFields
    advancedValues={Object {}}
    setAdvancedValues={[MockFunction]}
  >
    <Title
      className="advanced-fields-title"
      headingLevel="h2"
    >
      <h2
        className="pf-c-title pf-m-xl advanced-fields-title"
      >
        Advanced Fields
      </h2>
    </Title>
    <Form>
      <form
        className="pf-c-form"
        noValidate={true}
      >
        <EffectiveUserField
          setValue={[Function]}
          value=""
        >
          <mockConstructor
            fieldId="effective-user"
            label="Effective user"
            labelIcon={
              <Popover
                aria-label="help-text"
                bodyContent="A user to be used for executing the script. If it differs from the SSH user, su or sudo is used to switch the accounts."
                id="effective-user-help"
              >
                <button
                  aria-label="open-help-tooltip-button"
                  className="pf-c-form__group-label-help"
                  onClick={[Function]}
                >
                  <HelpIcon
                    color="currentColor"
                    noVerticalAlign={true}
                    size="sm"
                  />
                </button>
              </Popover>
            }
          >
            <div />
          </mockConstructor>
        </EffectiveUserField>
        <TimeoutToKillField
          setValue={[Function]}
          value=""
        >
          <mockConstructor
            fieldId="timeout-to-kill"
            label="Timeout to kill"
            labelIcon={
              <Popover
                aria-label="help-text"
                bodyContent="Time in seconds from the start on the remote host after which the job should be killed."
                id="timeout-to-kill-help"
              >
                <button
                  aria-label="open-help-tooltip-button"
                  className="pf-c-form__group-label-help"
                  onClick={[Function]}
                >
                  <HelpIcon
                    color="currentColor"
                    noVerticalAlign={true}
                    size="sm"
                  />
                </button>
              </Popover>
            }
          >
            <div />
          </mockConstructor>
        </TimeoutToKillField>
        <PasswordField
          setValue={[Function]}
          value=""
        >
          <mockConstructor
            fieldId="password"
            label="Password"
            labelIcon={
              <Popover
                aria-label="help-text"
                bodyContent="Password is stored encrypted in DB until the job finishes. For future or recurring executions, it is removed after the last execution."
                id="password-help"
              >
                <button
                  aria-label="open-help-tooltip-button"
                  className="pf-c-form__group-label-help"
                  onClick={[Function]}
                >
                  <HelpIcon
                    color="currentColor"
                    noVerticalAlign={true}
                    size="sm"
                  />
                </button>
              </Popover>
            }
          >
            <div />
          </mockConstructor>
        </PasswordField>
        <KeyPassphraseField
          setValue={[Function]}
          value=""
        >
          <mockConstructor
            fieldId="key-passphrase"
            label="Private key passphrase"
            labelIcon={
              <Popover
                aria-label="help-text"
                bodyContent="Key passphrase is only applicable for SSH provider. Other providers ignore this field. Passphrase is stored encrypted in DB until the job finishes. For future or recurring executions, it is removed after the last execution."
                id="key-passphrase-help"
              >
                <button
                  aria-label="open-help-tooltip-button"
                  className="pf-c-form__group-label-help"
                  onClick={[Function]}
                >
                  <HelpIcon
                    color="currentColor"
                    noVerticalAlign={true}
                    size="sm"
                  />
                </button>
              </Popover>
            }
          >
            <div />
          </mockConstructor>
        </KeyPassphraseField>
        <EffectiveUserPasswordField
          setValue={[Function]}
          value=""
        >
          <mockConstructor
            fieldId="effective-user-password"
            label="Effective user password"
            labelIcon={
              <Popover
                aria-label="help-text"
                bodyContent="Effective user password is only applicable for SSH provider. Other providers ignore this field. Password is stored encrypted in DB until the job finishes. For future or recurring executions, it is removed after the last execution."
                id="effective-user-password-help"
              >
                <button
                  aria-label="open-help-tooltip-button"
                  className="pf-c-form__group-label-help"
                  onClick={[Function]}
                >
                  <HelpIcon
                    color="currentColor"
                    noVerticalAlign={true}
                    size="sm"
                  />
                </button>
              </Popover>
            }
          >
            <div />
          </mockConstructor>
        </EffectiveUserPasswordField>
        <ConcurrencyLevelField
          setValue={[Function]}
          value=""
        >
          <mockConstructor
            fieldId="concurrency-level"
            label="Concurrency level"
            labelIcon={
              <Popover
                aria-label="help-text"
                bodyContent="Run at most N tasks at a time. If this is set and proxy batch triggering is enabled, then tasks are triggered on the smart proxy in batches of size 1."
                id="concurrency-level-help"
              >
                <button
                  aria-label="open-help-tooltip-button"
                  className="pf-c-form__group-label-help"
                  onClick={[Function]}
                >
                  <HelpIcon
                    color="currentColor"
                    noVerticalAlign={true}
                    size="sm"
                  />
                </button>
              </Popover>
            }
          >
            <div />
          </mockConstructor>
        </ConcurrencyLevelField>
        <TimeSpanLevelField
          setValue={[Function]}
          value=""
        >
          <mockConstructor
            fieldId="time-span"
            label="Time span"
            labelIcon={
              <Popover
                aria-label="help-text"
                bodyContent="Distribute execution over N seconds. If this is set and proxy batch triggering is enabled, then tasks are triggered on the smart proxy in batches of size 1."
                id="time-span-help"
              >
                <button
                  aria-label="open-help-tooltip-button"
                  className="pf-c-form__group-label-help"
                  onClick={[Function]}
                >
                  <HelpIcon
                    color="currentColor"
                    noVerticalAlign={true}
                    size="sm"
                  />
                </button>
              </Popover>
            }
          >
            <div />
          </mockConstructor>
        </TimeSpanLevelField>
      </form>
    </Form>
  </AdvancedFields>
</Provider>
`;
